src/Components/Assets/AssetFilter.tsx

```diff
@@ -38,12 +32,32 @@ function AssetFilter(props: any) {
   const [asset_class, setAssetClass] = useState<string>(
     filter.asset_class || ""
   );
-  const [facilityId, setFacilityId] = useState<number | "">(filter.facility);
+  const [facilityId, setFacilityId] = useState<string | "">(filter.facility);
   const [locationId, setLocationId] = useState<string | "">(filter.location);
   const [qParams, _] = useQueryParams();
 
+  const { data: facilityData } = useQuery(routes.getPermittedFacility, {
+    pathParams: { id: facilityId },
+  });
+
+  const { data: locationData } = useQuery(routes.getFacilityAssetLocation, {
+    pathParams: {
+      facilityId: String(facilityId),
+      locationId: String(locationId),
+    },
+  });
+
   useEffect(() => {
-    setFacilityId(facility?.id ? facility?.id : "");
+    if (facilityData) {
+      setFacility(facilityData);
+    }
+    if (locationData) {
+      setLocation(locationData);
+    }
+  }, [facilityData, locationData]);
```

Comment:
instead of putting these callbacks inside a `useEffect`, you can add callbacks to the `useQuery` options itself directly!

```
useQuery(route, {
  onResponse: ({ res, data }) => {
    if (res?.status === 200 && data) {
      setFacility(data);
    }
  }
})
```